executor/README.md

```diff
@@ -0,0 +1,270 @@
+| `async` | boolean | An optional flag. If set, the command runs asynchronously, in parallel with other future commands. It will be killed when all synchronous commands for a job are finished. |
+
+#### `GET /rawscheduler?job=:job`
+
+The job status endpoint can be used to view a job's progress, custom failure message, as well as the exit codes for all commands, by examining the following additional keys:
+
+| key | type    | description                              |
+| -------- | ------- | ---------------------------------------- |
+| `exit_code`     | integer | Status code when job specifies `command`  (will be absent if job has yet to finish). |
+| `before_exit_codes` | array of integers | Status codes for `before_commands`. |
+| `after_exit_codes` | array of integers | Status codes for `after_commands`. |
+| `progress` | long | Progress reported by a job's commands via the executor HTTP API. May be absent. |
+| `message` | string | A custom failure message set by a command using the executor HTTP API. May be absent. |
+
+### Executor HTTP API
+
+The executor provides an HTTP API that can be used from a job's commands. It currently only supports two endpoint:
+
+#### `PATCH /task/:taskid`
+
+This request also requires a body. Since it is a `PATCH` request, a valid request may contain only some of the following keys:
+
+| key | type   | description                              |
+| -------- | ------ | ---------------------------------------- |
+| `progress` | float  | A number between 0 and 100, representing the progress of the task |
+| `message`   | string | A custom failure reason, expected to be accompanied by a non-zero exit code. |
+| `env`      | object | A map of env variable names to values, a null value will unset the variable. This is really only useful from a before command, since it will only affect commands that run afterward.  |
+
+Example - progress update:
+
+`curl -X PATCH -d '{progress: 10}' $EXECUTOR_ENDPOINT`
+
+Example - setting and unsetting environment variables:
+
+`curl -X PATCH -d '{env: {VAR_A: "123", SOME_B: null}}' $EXECUTOR_ENDPOINT`
+
+Potential responses:
+
+| Code | HTTP Meaning        | Possible reason                          |
+| ---- | ------------------- | ---------------------------------------- |
+| 204  | No Content          | The request was succesfully processed. |
+| 400  | Malformed           | The request body contains an unknown key, or a key's value is the wrong type. |
+| 503  | Service Unavailable | The request was valid, and received, but there was an (ostensibly temporary) failure with mesos itself. |
+
+#### `GET /task/:taskid`
+
+For successful requests, the response body could contain:
+
+| key | type | description |
+|-----|------|-------------|
+| `exit_code` | int | The exit code of the primary command. |
+| `before_codes` | array of ints | The exit codes of the before commands. |
```

Comment:
Please see comment above regarding using an `id` for command. Then we can return a map instead of an array of exit codes.